test/gtest/kernel_tuning_net.cpp

```diff
@@ -181,11 +160,76 @@ class KernelTuningNetTest : public ::testing::TestWithParam<KernelTuningNetTestC
                                                               conv_desc,
                                                               test_case.direction);
 
-        Solver perf_config;
+        auto data_type = test_case.data_type;
+        auto in_tensor =
+            GPUMem{0, input_tensor_desc.GetNumBytes() / sizeof(data_type), sizeof(data_type)};
+        auto wt_tensor =
+            GPUMem{0, weights_tensor_desc.GetNumBytes() / sizeof(data_type), sizeof(data_type)};
+        auto out_tensor =
+            GPUMem{0, output_desc.GetNumBytes() / sizeof(data_type), sizeof(data_type)};
+        auto workSpaceSize = conv_desc.GetWorkSpaceSize(ctx, problem);
+        auto workSpace     = GPUMem{0, workSpaceSize / sizeof(data_type), sizeof(data_type)};
```

Comment:
Because of splitting bytes into elements, some solvers require workspace to be aligned like (128B, 256B...etc) - or have a multiple of some tile size. 

So it might be a good idea to assert it meets alignment:

`assert((workSpaceSize % required_alignment) == 0);`